src/h2_stream.erl

```diff
@@ -183,6 +202,19 @@ callback_mode() ->
 %% PUSH_PROMISE frame with that Stream Id. It's a subtle thing, but it
 %% drove me crazy until I figured it out
 
+callback(undefined, _, _, State) ->
+    {ok, State};
+callback(Mod, Fun, Args, State) ->
+    %% load the module if it isn't already
+    AllArgs = Args ++ [State],
+    erlang:function_exported(Mod, module_info, 0) orelse code:ensure_loaded(Mod),
+    case erlang:function_exported(Mod, Fun, length(AllArgs)) of
+        true ->
+            erlang:apply(Mod, Fun, AllArgs);
+        false ->
+            {ok, State}
```

Comment:
not sure why this isn't an error? in what case would a callback function specified not exist but not be an error?

Reply:
The user doesn't have to support every callback from h2_stream. Maybe they should, or at least some f them, but I just went with not requiring any stream handling callback being required from the user.

Reply:
@talentdeficit ping